### PR TITLE
chore(portal): Fix resources UX

### LINKED ISSUE #3132: Resources UX improvements
- [x] `Add multi-site Resource` button feature flag
- [ ] ~Change text when you remove Resource to say "Remove from Site"~
- [x] Bring back sidebar Resources
- [x] Don't return Resources with no Sites to client

### PR DESCRIPTION
**Pull request by AndrewDryga**
Closes #3132

### CODE REVIEW

File: elixir/apps/domain/test/domain/resources_test.exs [hunk 1]
```diff
@@ -471,6 +499,35 @@ defmodule Domain.ResourcesTest do
       assert length(resources) == 2
     end
 
+    test "does not authorized resources for deleted gateway groups", %{
```
Comment (jamilbk) [inline elixir/apps/domain/test/domain/resources_test.exs]:
```suggestion
    test "does not authorize resources for deleted gateway groups", %{
```

File: elixir/apps/web/lib/web/live/resources/index.ex [hunk 2]
```diff
@@ -33,8 +33,11 @@ defmodule Web.Resources.Index do
         Resources
       </:title>
       <:action>
-        <.add_button navigate={~p"/#{@account}/resources/new"}>
-          Add Resource
+        <.add_button
+          :if={Domain.Config.multi_site_resources_enabled?()}
+          navigate={~p"/#{@account}/resources/new"}
+        >
+          Add Multi-Site Resource
```
Comment (jamilbk) [inline elixir/apps/web/lib/web/live/resources/index.ex:40]:
```suggestion
          Add Multi-site Resource
```

File: elixir/apps/web/test/web/live/resources/index_test.exs [hunk 3]
```diff
@@ -49,7 +49,23 @@ defmodule Web.Live.Resources.IndexTest do
...
+  } do
+    Domain.Config.feature_flag_override(:multi_site_resources, false)
+
+    {:ok, _lv, html} =
+      conn
+      |> authorize_conn(identity)
+      |> live(~p"/#{account}/resources")
+
+    assert button = Floki.find(html, "a[href='/#{account.slug}/resources/new']")
+    refute Floki.text(button) =~ "Add Multi-Site Resource"
```
Comment (jamilbk) [inline elixir/apps/web/test/web/live/resources/index_test.exs:68]:
```suggestion
    refute Floki.text(button) =~ "Add Multi-site Resource"
```

File: elixir/apps/web/test/web/live/resources/index_test.exs [hunk 4]
```diff
@@ -49,7 +49,23 @@ defmodule Web.Live.Resources.IndexTest do
       |> live(~p"/#{account}/resources")
 
     assert button = Floki.find(html, "a[href='/#{account.slug}/resources/new']")
-    assert Floki.text(button) =~ "Add Resource"
+    assert Floki.text(button) =~ "Add Multi-Site Resource"
```
Comment (jamilbk) [inline elixir/apps/web/test/web/live/resources/index_test.exs:52]:
```suggestion
    assert Floki.text(button) =~ "Add Multi-site Resource"
```

**Review by jamilbk — COMMENTED** [review]
~~Just some minor capitalization comments.~~ Edit: I was wrong, apparently it's common to Camel-Case hyphenated words too. Thanks chatgpt

I'll need to document / market this really well as I think it's a neat feature. Did you have an obvious use-case in mind I could lay out in the docs?

Thanks for making it disabled by default (I think without docs and a little more feedback it might be confusing atm)

**Comment by AndrewDryga** [comment]
> I'll need to document / market this really well as I think it's a neat feature. Did you have an obvious use-case in mind I could lay out in the docs?

Right now we just want to provide a way to delete a resource after you delete its site. 

As a use case for documentation, we can list pretty much any case where you have two replicas of exactly the same thing running across multiple regions. A remote shell into our relay VMs would be a case where we do that. More direct use cases:

1. Geographically distributed resources (eg. applications);
2. Multi-cloud Kubernetes;
3. Useful during migration between two clouds.

**Comment by bmanifold** [comment]
Found one small thing.  I noticed that the `edit` page for a multi-site resource won't allow you to save a resource if no sites are selected, but there doesn't seem to be an error message displayed as to why it's not saved.

**Comment by AndrewDryga** [comment]
@bmanifold fixed.

**Review by jamilbk — APPROVED** [review]

**Pull request merged**